internal/telemetry/segment.go

```diff
@@ -0,0 +1,147 @@
+package telemetry
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"log"
+	"math"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/samber/lo"
+	segment "github.com/segmentio/analytics-go"
+	"go.jetpack.io/devbox/internal/build"
+	"go.jetpack.io/devbox/internal/cloud/openssh"
+)
+
+// cmdStartTime records the time at the start of any devbox command invocation.
+var cmdStartTime time.Time
+
+// Event contains common fields used in our segment events
+type Event struct {
+	AnonymousID string
+	AppName     string
+	AppVersion  string
+	CloudRegion string
+	OsName      string
+	UserID      string
+}
+
+// ttiEvent contains fields used to log the time-to-interactive event. For now,
+// this is used for devbox shell (local and cloud).
+type ttiEvent struct {
+	Event
+	// TODO savil. Can this be time.Duration as done in midcobra.telemetry?
+	durationSeconds int
+}
+
+// NewSegmentClient returns a client object to use for segment logging.
+// Callers are responsible for calling client.Close().
+func NewSegmentClient(telemetryKey string) segment.Client {
+	segmentClient, _ := segment.NewWithConfig(telemetryKey, segment.Config{
+		BatchSize: 1, /* no batching */
+		// Discard logs:
+		Logger:  segment.StdLogger(log.New(io.Discard, "" /* prefix */, 0)),
+		Verbose: false,
+	})
+
+	return segmentClient
+}
+
+// CommandStartTime records and returns the time at the start of the command invocation.
+// It must be called initially at the start of the cobra (or other framework) command
+// stack. Subsequent calls returns the time from the first invocation of this function.
+func CommandStartTime() time.Time {
+	if cmdStartTime.IsZero() {
+		cmdStartTime = time.Now()
+	}
+	return cmdStartTime
+}
+
+// LogShellTimeToInteractiveEvent logs the duration from start of the command
+// till the shell was ready to be interactive.
+func LogShellTimeToInteractiveEvent(startTime string) error {
+	start, err := timeFromUnixTimestamp(startTime)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	evt := ttiEvent{
+		Event: Event{
+			AnonymousID: DeviceID(),
+			AppName:     "",
```

Comment:
Why `""`?

Reply:
oh oops, I need to hook up AppName and AppVersion. Will fix!